homeassistant/components/elmax/config_flow.py

```diff
@@ -0,0 +1,218 @@
+"""Config flow for elmax-cloud integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from elmax_api.exceptions import ElmaxBadLoginError, ElmaxBadPinError, ElmaxNetworkError
+from elmax_api.http import Elmax
+from elmax_api.model.panel import PanelEntry
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.elmax.const import (
+    CONF_ELMAX_PANEL_ID,
+    CONF_ELMAX_PANEL_NAME,
+    CONF_ELMAX_PANEL_PIN,
+    CONF_ELMAX_PASSWORD,
+    CONF_ELMAX_USERNAME,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_IGNORE, SOURCE_USER, ConfigEntry
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+_LOGGER = logging.getLogger(__name__)
+
+
+LOGIN_FORM_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ELMAX_USERNAME): str,
+        vol.Required(CONF_ELMAX_PASSWORD): str,
+    }
+)
+
+
+def _store_panel_by_name(
+    panel: PanelEntry, username: str, panel_names: dict[str, str]
+) -> None:
+    original_panel_name = panel.get_name_by_user(username=username)
+    panel_id = panel.hash
+    collisions_count = 0
+    panel_name = original_panel_name
+    while panel_name in panel_names:
+        # Handle same-name collision.
+        collisions_count += 1
+        panel_name = f"{original_panel_name} ({collisions_count})"
+    panel_names[panel_name] = panel_id
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for elmax-cloud."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
```

Comment:
The connection class should be moved to the manifest file nowadays.

Reply:
You're right, that was redundant: the current manifest already included the iot_class attribute. I've removed line 55, which was redundant.